fem/fespace.cpp

```diff
@@ -4497,6 +4499,160 @@ ::GetEssentialBdrEdgesFaces(const Array<int> &bdr_attr_is_ess,
    }
 }
 
+void FiniteElementSpace::GetBoundaryEdgeDofs(
+   const Array<int> &boundary_element_indices,
+   std::unordered_set<int> &boundary_edge_dofs,
+   std::unordered_map<int, int> &dof_to_edge,
+   std::unordered_map<int, int> &dof_to_boundary_element,
+   std::unordered_map<int, int> &dof_to_edge_orientation) const
+{
+   MFEM_VERIFY(mesh->Dimension() >= 3,
+               "GetBoundaryEdgeDofs requires 3D meshes to find 1D edge objects");
+
+   // Clear output containers
+   boundary_edge_dofs.clear();
+   dof_to_edge.clear();
+   dof_to_boundary_element.clear();
+   dof_to_edge_orientation.clear();
+
+   // Collect boundary edge DoFs using a toggle approach
+   Array<int> edges, edge_orientations, edge_dofs;
+   for (int i = 0; i < boundary_element_indices.Size(); ++i)
+   {
+      int boundary_element_idx = boundary_element_indices[i];
+      int face_index, face_orientation;
+      mesh->GetBdrElementFace(boundary_element_idx, &face_index, &face_orientation);
+      mesh->GetFaceEdges(face_index, edges, edge_orientations);
+
+      for (int j = 0; j < edges.Size(); ++j)
+      {
+         GetEdgeDofs(edges[j], edge_dofs);
+         for (int k = 0; k < edge_dofs.Size(); ++k)
+         {
+            int dof = edge_dofs[k];
+            if (!boundary_edge_dofs.count(dof))
+            {
+               boundary_edge_dofs.insert(dof);
+               dof_to_edge[dof] = edges[j];
+               dof_to_boundary_element[dof] = boundary_element_idx;
+               dof_to_edge_orientation[dof] = edge_orientations[j];
+            }
+            else
+            {
+               // DoF appears twice - interior to boundary, remove it
```

Comment:
It appears this function only finds edges on the boundary of the boundary. For example, if there is a square cut on the interior of a 3D domain, this function will find edges on the 4 sides of the interior square. If the interior boundary is the surface of a cube inside the domain (like the plot shown in the PR description), it will not find any edges. Is this right? In any case, more documentation for this function in the header file would be helpful, to explain exactly what it does.